include/RAJA/exec-cuda/MemUtils_CUDA.hxx

```diff
@@ -63,12 +64,16 @@ namespace RAJA {
 
 #define RAJA_CUDA_REDUCE_BLOCK_LENGTH (1024 + 8) * 16
 
+
 ///
 /// Typedef defining common data type for RAJA-Cuda reduction data blocks
 /// (use this in all cases to avoid type confusion).
 ///
 typedef double CudaReductionBlockDataType;
 
+typedef struct{double val; Index_type idx;} CudaReductionLocBlockDataType; 
```

Comment:
Yes, I struggled with that as well -- you nailed it with your comment "realizing that the largest will determine the size ..."  Here the assumption was that double was the largest type encountered